src/Tools/dotnet-dump/ExtensionCommands/ClrMDHelper.cs

```diff
@@ -271,29 +88,28 @@ public IEnumerable<TimerInfo> EnumerateTimers()
             {
                 // until the ClrMD bug to get static field value is fixed, iterate on each object of the heap
                 // to find each TimerQueue and iterate on (slower but it works)
-                foreach (var address in _heap.EnumerateObjectAddresses())
+                foreach (var obj in _heap.EnumerateObjects())
                 {
-                    var objType = _heap.GetObjectType(address);
+                    var objType = obj.Type;
                     if (objType == null) continue;
-                    if (objType == _heap.Free) continue;
+                    if (objType == _heap.FreeType) continue;
                     if (string.Compare(objType.Name, "System.Threading.TimerQueue", StringComparison.Ordinal) != 0)
                         continue;
 
                     // m_timers is the start of the linked list of TimerQueueTimer
-                    var currentTimerQueueTimer = GetFieldValue(address, "m_timers");
-
+                    var currentTimerQueueTimer = obj.ReadObjectField("m_timers");
+                
                     // iterate on each TimerQueueTimer in the linked list
-                    while ((currentTimerQueueTimer != null) && (((ulong)currentTimerQueueTimer) != 0))
+                    while (currentTimerQueueTimer.Address != 0)
```

Comment:
FYI, ClrObject implements an `implicit operator ulong` for .Address.  If you need to pass a ClrObject to a method or property accepting a ulong, you can pass it directly instead of .Address.  https://github.com/microsoft/clrmd/blob/master/src/Microsoft.Diagnostics.Runtime/src/Common/ClrObject.cs#L216

So code like this can just be `currentTimeQueueTimer != 0` without the `.Address`.  I'm just pointing this out as a general FYI, not actual code review feedback to change here.  If you prefer this style that's fine too.

Reply:
Similar with `ti.StateAddress = state.Address;` below.

Reply:
I'll let @chrisnas change his code in a future PR.